internal/pkg/db/mongo/export.go

```diff
@@ -176,33 +151,28 @@ func (mc MongoClient) getRegistration(q bson.M) (models.Registration, error) {
 
 	var reg models.Registration
 	err := s.DB(mc.database.Name).C(db.ExportCollection).Find(q).One(&reg)
-	if err == mgo.ErrNotFound {
-		return models.Registration{}, db.ErrNotFound
+	if err != nil {
+		return models.Registration{}, errorMap(err)
 	}
-
-	return reg, err
+	return reg, nil
 }
 
 // Delete from the collection based on ID
 func (mc MongoClient) deleteRegistration(q bson.M) error {
 	s := mc.getSessionCopy()
 	defer s.Close()
 
-	err := s.DB(mc.database.Name).C(db.ExportCollection).Remove(q)
-	if err == mgo.ErrNotFound {
-		return db.ErrNotFound
-	}
-	return err
+	return errorMap(s.DB(mc.database.Name).C(db.ExportCollection).Remove(q))
 }
 
 func mapRegistrations(registrations []models.Registration, err error) ([]contract.Registration, error) {
 	if err != nil {
 		return []contract.Registration{}, err
 	}
 
-	var mapped []contract.Registration
+	mapped := make([]contract.Registration, 0)
```

Comment:
Is there some reason why this is the preferred format for this?

Reply:
In this particular case, when `registrations` is empty `make` results in an empty slice and `var` results in nil.  This specific change was made because convention is to return an empty slice instead of nil.

If the question is why use `make([]contract.Registration, 0)` in place of `[]contract.Registration{}`, it's to address the `Empty slice declaration via literal` warning in GoLand.

Reply:
Thanks

Reply:
@tsconn23 His point was that if append isn't called at all then `nil` will be returned, which is not what most of the methods have been doing in this case (returning empty slices).  I agree that `nil` should be treated as an empty slice by the caller, but to Michael's point the consistency might be worth it here.

Reply:
@tsconn23 I think I'm missing something here.  If `append()` is not called, using `var` leaves `mapped` nil.

Reply:
ah, got it. Yes, I was just focused on the append operation. Makes sense.